fvm/src/executor/mod.rs

```diff
@@ -43,18 +44,23 @@ impl ApplyRet {
         message: impl Into<String>,
         miner_penalty: BigInt,
     ) -> ApplyRet {
+        // This is a little funky, but it's the simplest way to record a sane error message.
+        let mut bt = Backtrace::default();
+        bt.set_cause(Cause {
+            module: "system",
+            function: "execute",
+            error: ErrorNumber::IllegalOperation,
```

Comment:
This is a bit weird, as we're picking this error from the syscall namespace?

Reply:
Yeah, I agree. I think the answer is to introduce a new type:

```rust
enum MessageFailure {
     PreValidation(...),
     Failure(Backtrace),
}
```
```

Reply:
How does it look now?